src/frequenz/sdk/timeseries/formula_engine/_formula_steps.py

```diff
@@ -390,7 +439,41 @@ async def fetch_next(self) -> Sample[QuantityT] | None:
         Returns:
             The fetched Sample.
         """
-        self._next_value = await self._stream.receive()
+        if (
+            self._fallback_metric_fetcher is not None
+            and self._fallback_metric_fetcher.is_running()
+        ):
+            tasks = [self._stream.receive(), self._fallback_metric_fetcher.fetch_next()]
+            result = await asyncio.gather(*tasks, return_exceptions=False)
+
+            # Synchronize the timestamps from two sources
+            while result[0].timestamp < result[1].timestamp:
+                result[0] = await self._stream.receive()
+
+            while result[0].timestamp > result[1].timestamp:
+                result[1] = await self._fallback_metric_fetcher.fetch_next()
+
+            if self._value_is_invalid(result[0].value):
+                self._next_value = result[1]
+            else:
+                self._next_value = result[0]
+                # We can stop the fallback metric fetcher if we have a valid value from
+                # primary component.
+                # This is not tested yet.
```

Comment:
Probably not to put in a code comment? (the "not tested yet")

Reply:
Probably best to drop the stop method completely and create an issue about it instead.

Reply:
Done :) 
https://github.com/frequenz-floss/frequenz-sdk-python/issues/1045